app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiLabOrderConverter.java

```diff
@@ -167,8 +170,24 @@ private ServiceRequest createServiceRequest(final Patient patient) {
 
         serviceRequest.setSubject(new Reference(patient));
 
-        serviceRequest.setAuthoredOn(Date.from(Instant.now()));
+        serviceRequest.setAuthoredOn(orderDate);
 
         return serviceRequest;
     }
+
+    private Provenance createProvenanceResource(Date orderDate) {
+        logger.logInfo("Creating new Provenance");
+        var provenance = new Provenance();
+
+        provenance.setId(UUID.randomUUID().toString());
+        provenance.setRecorded(orderDate);
+        provenance.setActivity(
+                new CodeableConcept(
+                        new Coding(
+                                "http://terminology.hl7.org/CodeSystem/v2-0003",
+                                "021",
+                                "OML - Laboratory order")));
```

Comment:
I see some duplication from the `createMessageHeader` method and this code.  I bet this `Coding` could be created once and used in both places.

Reply:
This should be refactored appropriately now... @halprin